### PR TITLE
Move joined pl sections to my pl page

### PR DESCRIPTION
**Pull request by TurnerRiley**
Moves the joined Professional Learning sections table to the My PL page!

### My PL page
With no PL sections:
My PL with no sections

With PL sections:
My PL with sections new

Joining a section shows success message:
New success

### Dashboard
Student view of joined sections (should not show anything PL related):
Just student sections on homepage

Teacher view shows "Join Section" and skip link to My PL page:
Teacher Dashboard no sections

## Links
Jira ticket: [here](https://codedotorg.atlassian.net/jira/software/c/projects/ACQ/boards/64?assignee=60d62161f65054006980bd71&selectedIssue=ACQ-1428)
Spec: [here](https://docs.google.com/document/d/1Ee02kw_ei2x6wOS06K768dC-kUhGDlydjoE6BZipbPE/edit#heading=h.99hp5m1b10li)
Figma: [here](https://www.figma.com/file/sZ6d1uhJWFksJ6j3Fnmbnc/PL-Flow-Improvements?type=design&node-id=2044-12349&mode=design&t=2LTRK8QYkD9Dzrp8-0)

## Testing story
Local testing and updating unit tests.

### CODE REVIEW

File: apps/src/templates/studioHomepages/ParticipantSections.jsx [hunk 1]
```diff
@@ -75,51 +76,56 @@ export default class ParticipantSections extends Component {
...
-                </div>
-                <div style={styles.hiddenSectionDesc}>
-                  {i18n.archivedSectionsStudentDescription()}
-                </div>
-                <SectionsAsStudentTable
-                  sections={archivedSections}
-                  canLeave={!!isTeacher}
-                  updateSections={this.props.updateSections}
-                  updateSectionsResult={this.props.updateSectionsResult}
+      <div style={styles.participantSectionsContainer}>
```
Comment (TurnerRiley) [inline apps/src/templates/studioHomepages/ParticipantSections.jsx:79]: This whole changed section is just from wrapping it all in a `div`, but Git is highlighting it weird

File: apps/src/templates/studioHomepages/SectionsAsStudentTable.jsx [hunk 2]
```diff
@@ -23,6 +24,25 @@ class SectionsAsStudentTable extends React.Component {
     isPlSections: PropTypes.bool,
   };
 
+  renderSectionCodeCell(section) {
```
Comment (TurnerRiley) [inline apps/src/templates/studioHomepages/SectionsAsStudentTable.jsx:27]: Made this function to pull the complicated terniary logic used below

File: apps/src/templates/studioHomepages/SectionsAsStudentTable.jsx [hunk 3]
```diff
@@ -52,33 +75,41 @@ class SectionsAsStudentTable extends React.Component {
       >
         <thead>
           <tr style={styles.headerRow}>
-            <td
+            <th
```
Comment (TurnerRiley) [inline apps/src/templates/studioHomepages/SectionsAsStudentTable.jsx:78]: These cells are all in the header row so they should be `th` tags, this also helps make styling easier

File: dashboard/app/controllers/pd/professional_learning_landing_controller.rb [hunk 4]
```diff
@@ -42,6 +42,8 @@ def index
       workshops_for_regional_partner: workshops_for_regional_partner_data,
       pl_courses_started: current_user.pl_units_started,
       user_permissions: current_user.permissions.map(&:permission),
+      joined_student_sections: current_user.sections_as_student_participant&.map(&:summarize_without_students),
```
Comment (bethanyaconnor) [inline dashboard/app/controllers/pd/professional_learning_landing_controller.rb:45]: woops sorry for not adding these before! Thanks for adding them now :)

File: apps/src/templates/studioHomepages/ParticipantSections.jsx (hunk 1 above)
Comment (bethanyaconnor) [reply apps/src/templates/studioHomepages/ParticipantSections.jsx:79]: Hiding whitespace helped a lot here!

**Review by bethanyaconnor — APPROVED** [review]
🎉 

I think at some point we discussed showing an error message or something if the user tries to join a non-PL section from the PL page. Can you make a ticket to add that? I'm a bit curious if anyone actually hits that case before we add it in.

**Pull request merged**